platooning_strategic_IHP/src/platoon_strategic_ihp.cpp

```diff
@@ -1168,14 +1615,95 @@ namespace platoon_strategic_ihp
             return MobilityRequestResponse::NO_RESPONSE;
         }
     }
+    
+    // UCLA: add request call-back function for lead with operation state (for cut-in join)
+    MobilityRequestResponse PlatoonStrategicIHPPlugin::mob_req_cb_leadwithoperation(const cav_msgs::MobilityRequest& msg)
+    {
+        /*
+        *   Current leader leading while opening the gap for cut-in join.
+        */
+
+        // Check request plan type  
+        cav_msgs::PlanType plan_type = msg.plan_type;
+        if (plan_type.type == cav_msgs::PlanType::PLATOON_CUTIN_JOIN) 
+        {
+            // task 1. slow down to create gap 
+            int join_index = msg.join_index;
+            // determine if joining vehicle in position
+            double cutinDtdDifference = current_downtrack_ - pm_.platoon[0].vehiclePosition - config_.vehicleLength;
```

Comment:
Not clear what is being computed. This method is for when host is platoon leader that someone else is joining, right?  Then current_downtrack_ is the host's DTD, which is the same as pm_.platoon[0].vehicle_position, right?  Therefore, cutinDtDDifference will always be -vehicleLength.  I may be seeing something wront.